yacc.go

```diff
@@ -3331,18 +3323,17 @@ type $$Parser interface {
 }
 
 type $$ParserImpl struct {
-	lookahead func() int
+	lval  $$SymType
+	stack [16]$$SymType
```

Comment:
it seems to me that the performance gain comes from reusing the slice, and not necessarily from making it smaller.